File: _wpf/wpf-business-programmer/3-wpf-business-programmer-m3-exercise-files/Samples-2012/CSharp/WPFTemplateApp/WPFTemplateApp/Common/ObjectBase.cs
using System.ComponentModel;
using System.Windows;

namespace WPFTemplateApp
{
  public class ObjectBase : DependencyObject, INotifyPropertyChanged
  {
    #region INotifyPropertyChanged
    public event PropertyChangedEventHandler PropertyChanged;

    protected void RaisePropertyChanged(string propertyName)
    {
      if (PropertyChanged != null)
        PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
    }
    #endregion
  }
}

File: _wpf/wpf-business-programmer/3-wpf-business-programmer-m3-exercise-files/Samples-2012/VB/WPFTemplateApp/WPFTemplateApp/Common/ObjectBase.vb
Imports System.ComponentModel
Imports System.Windows

Public Class ObjectBase
  Inherits DependencyObject
  Implements INotifyPropertyChanged

#Region "INotifyPropertyChanged"
  Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

  Protected Sub RaisePropertyChanged(ByVal propertyName As String)
    RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
  End Sub
#End Region
End Class
